modules/layers/src/geojson-layer/geojson-layer-props.js

```diff
@@ -95,10 +94,13 @@ export function createLayerPropsFromBinary(geojsonBinary, uniqueIdProperty, enco
   layerProps.polygons._normalize = false;
   if (polygons.triangles) {
     layerProps.polygons.data.attributes.indices = polygons.triangles.value;
+    layerProps.polygons.data.length = polygons.polygonIndices.value.length - 1;
```

Comment:
Why is the count of polygons dependent on triangulation?

Reply:
Prior to triangulation the length includes the polygons which define holes (in the old code path before we added triangulation in the worker). If we've already triangulated, and thus have the indices attribute we need to set the data.length based on the number of polygon objects (where each object could contain holes)

Reply:
Triangulated or not it doesn't change the `properties` array. If `data.length` does not match the original polygon count in the feature collection then you'll end up looking for feature indices that do not exist.

Reply:
I was going by what the old code had, but from testing it appears that the old path had it wrong. I've fixed it so that the length is based on the number of features, which is given by the `polygonIndices.length`. The added tests confirm this